src/metrics.jl

```diff
@@ -236,3 +236,45 @@ function avg_gate_fidelity(x::T, y::T) where T <: Union{PauliTransferMatrix{B, B
     dim = 2 ^ length(x.basis_l)
     return (tr(transpose(x.data) * y.data) + dim) / (dim^2 + dim)
 end
+
+"""
+    entanglement_entropy(state, partition, [entropy_fun=entropy_vn])
+
+Computes the entanglement entropy of `state` between the list of sites `partition`
+and the rest of the system. The state must be defined in a composite basis.
+
+By default the entropy is computed according to Von-Neumann entropy, but a different
+function can be provided (for example to compute the entanglement-reny entropy).
+"""
+function entanglement_entropy(psi::Ket{B}, partition, entropy_fun=entropy_vn) where B<:CompositeBasis
```

Comment:
The `if partition isa Number` can be handled more nicely by a dispatch. Add type constraints here `partition::Vector{Int}`. 
Then, another method below like this `entanglement_entropy(state, partition::Number, args...) = entanglement_entropy(state, Int[partition], args...)` deals with number partitions.